### DISCUSSION TITLE
Error since updating from version 7.2.0 to 7.3.0 in next js app

### DISCUSSION THREAD

**Discussion by benzler** [wojtekmaj/react-pdf#1577] [Q&A] [answered]
Since upgrading to the newest version I have the following error in my typescript next js app:

```
SyntaxError: Named export 'PDFDataRangeTransport' not found. The requested module 'pdfjs-dist' is a CommonJS module, which may not support all module.exports as named exports.
```

This is my code to show the pdf:
``` 

import { Document, Page, pdfjs } from 'react-pdf';
import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
import 'react-pdf/dist/esm/Page/TextLayer.css';


  useEffect(() => {
    if (typeof window !== 'undefined') {
      // https://stackoverflow.com/questions/65740268/create-react-app-how-to-copy-pdf-worker-js-file-from-pdfjs-dist-build-to-your
      pdfjs.GlobalWorkerOptions.workerSrc = new URL(
        'pdfjs-dist/build/pdf.worker.min.js',
        import.meta.url,
      ).toString();
    }
  }, []);

  const [numPages, setNumPages] = useState(null);
  const [pageNumber, setPageNumber] = useState(1);

  function onDocumentLoadSuccess({ numPages }: { numPages: number }) {
    setNumPages(numPages);
  }

 <Document
                      file={filePreview}
                      onLoadSuccess={onDocumentLoadSuccess}
                    >
                      <Page pageNumber={pageNumber} />
                    </Document>
```

Can please someone help me how to fix this error?

Before updating the code worked perfectly.

Seems like I'm not the only one having this issue: https://stackoverflow.com/questions/76796240/named-export-pdfdatarangetransport-not-found-while-using-react-pdf-in-next-js?newreg=704b67926da2412d96aa0144a5a69bd6

**Comment by wojtekmaj** [comment, marked as answer]
Yeah, until 7.3.0 React-PDF was not supporting ESM modules natively. It now does, but it comes with some unexpected problems with strictness (which, when resolved, will only benefit everyone). 

Definitely not your fault - will look into it as soon as possible. Thanks for the heads up.

**Reply by benzler** [reply]
alright, thanks for your fast response!

**Reply by wojtekmaj** [reply]
Fixed in v7.3.1, but there's still another bug that may happen in Next.js for some reason. Still investigating.

**Reply by wojtekmaj** [reply]
For anyone curious - we got a series of fixes merged, ending up with v7.3.3 which seems to be pretty stable.

**Reply by sulemankhan1** [reply]
Thanks it was a headache to solve this issue your comment saved my Day :D
